cmd/zc_traverser_blob.go

```diff
@@ -71,14 +71,25 @@ type blobTraverser struct {
 	isDFS bool
 }
 
-func (t *blobTraverser) IsDirectory(isSource bool) (bool, error) {
+var NonErrorDirectoryStubOverlappable = errors.New("The directory stub exists, and can overlap.")
+
+func (t *blobTraverser) IsDirectory(isSource bool) (isDirectory bool, err error) {
 	isDirDirect := copyHandlerUtil{}.urlIsContainerOrVirtualDirectory(t.rawURL)
 
 	// Skip the single blob check if we're checking a destination.
 	// This is an individual exception for blob because blob supports virtual directories and blobs sharing the same name.
 	// On HNS accounts, we would still perform this test. The user may have provided directory name without path-separator
 	if isDirDirect { // a container or a path ending in '/' is always directory
-		return true, nil
+		// If it's a container, let's ensure that container exists. Listing is a safe assumption to be valid, because how else would we enumerate?
+		blobURLParts, err := blob.ParseURL(t.rawURL)
```

Comment:
Do we also want this to hit if the URL is a virtual directory? I think isDirDirect could also be true for virtual directories.

Reply:
Good catch, this was lazy 4 am coding; I thought about this directly and then proceeded to not check.

Reply:
No, this was not. If it is a vdir, we should still validate the container exists.